### PR TITLE
Feature/ml 200 aapl on automl when i use trainpredict image classif

### PR DESCRIPTION
**Pull request by crsegerie**

### CODE REVIEW

File: kiliautoml/utils/helpers.py [hunk 1]
```diff
@@ -137,6 +132,7 @@ def get_assets(
     max_assets: Optional[int] = None,
     labeling_statuses: List[labeling_statusT] = ["LABELED", "UNLABELED"],
 ) -> List[Dict]:
+    kili_print("Downloading datasets metadata from Kili")
```
Comment (PierreLeveau) [inline kiliautoml/utils/helpers.py:135]: "Downloading asset metadata from Kili"

File: label_errors.py [hunk 2]
```diff
@@ -110,19 +87,28 @@ def download_assets(project_id, assets, api_key, data_path, job_name):
 )
 @click.option("--project-id", default=None, required=True, help="Kili project ID")
 @click.option("--verbose", default=0, type=int, help="Verbose level")
+@click.option(
+    "--disable-wandb",
+    default=False,
```
Comment (PierreLeveau) [inline label_errors.py:92]: default = True for this one?

File: kiliautoml/utils/helpers.py [hunk 3]
```diff
@@ -217,59 +213,8 @@ def get_last_trained_model_path(
...
-                externalId=asset["externalId"],
-                filename=filename or "",
-                image=image,
-            )
-        )
-    return downloaded_images
-
-
-def save_errors(found_errors, job_path):
+# TODO: This function is not used anywhere
```
Comment (PierreLeveau) [inline kiliautoml/utils/helpers.py:216]: rm

File: kiliautoml/utils/type.py [hunk 4]
```diff
@@ -3,3 +3,4 @@
 labeling_statusT = Literal["LABELED", "UNLABELED"]
 status_inT = Literal["TODO", "ONGOING", "LABELED", "REVIEWED", "DEFAULT"]
 label_typeT = Literal["DEFAULT", "REVIEW"]
+commandT = Literal["train", "predict", "label_errors", "prioritize"]
```
Comment (PierreLeveau) [inline kiliautoml/utils/type.py:6]: types should be CamelCase (did not catch that earlier sorry)

File: prioritize.py [hunk 5]
```diff
@@ -437,7 +439,7 @@ def main(
     )
 
     if input_type == InputType.Image:
-        downloaded_images = download_project_images(project_id, api_key, unlabeled_assets)
+        downloaded_images = download_project_images(api_key, unlabeled_assets)
```
Comment (PierreLeveau) [inline prioritize.py:442]: `project_id` should be kept, it is now required in the request header

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: label_errors.py (hunk 2 above)
Comment (crsegerie) [reply label_errors.py:92]: hum, maybe for label error we can put it to false

**Comment by crsegerie** [comment]
commit [94b6d7f] is pass the tests locally. Not enough memory on github action

File: kiliautoml/utils/cleanlab/datasets.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import copy
+from typing import Dict
+
+import torch.utils.data as torch_Data
+from sklearn.model_selection import train_test_split
+from torchvision import datasets, transforms
+from typing_extensions import Literal
+
+training_phaseT = Literal["train", "val"]
```
Comment (PierreLeveau) [inline kiliautoml/utils/cleanlab/datasets.py:9]: TrainingPhaseT

File: kiliautoml/utils/cleanlab/datasets.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+dict_datasetT = Dict[training_phaseT, torch_Data.Dataset]
```
Comment (PierreLeveau) [inline kiliautoml/utils/cleanlab/datasets.py:10]: DictDatasetT

File: kiliautoml/utils/path.py [hunk 8]
```diff
@@ -1,45 +1,122 @@
...
-    @staticmethod
-    def inference(root_dir, project_id, job_name, model_repository: ModelRepositoryT):
-        return os.path.join(root_dir, project_id, job_name, model_repository, "inference")
 
-    # those 3 functions use the path value for huggingface.
+"""
+Once we have the model repository dir, we can create the following nested directories:
+"""
+
+ModelDirT = str
```
Comment (PierreLeveau) [inline kiliautoml/utils/path.py:70]: move up with the other types

File: train.py [hunk 9]
```diff
@@ -140,19 +154,22 @@ def train_image_bounding_box(
...
 )
+@click.option("--verbose", default=0, type=int, help="Verbose level")
 def main(
     api_endpoint: str,
     api_key: str,
     model_framework: ModelFrameworkT,
     model_name: ModelNameT,
     model_repository: ModelRepositoryT,
     project_id: str,
+    epochs: int,
```
Comment (PierreLeveau) [inline train.py:165]: if we add epochs, it must be bonded to all the other training models

File: train.py [hunk 10]
```diff
@@ -242,6 +263,33 @@ def main(
...
+
+            assets = get_assets(
+                kili,
+                project_id,
+                parse_label_types(label_types),
+                labeling_statuses=["LABELED"],
+            )
+            assets = assets[:max_assets] if max_assets is not None else assets
+
+            training_loss = train_and_get_error_image_classification(
```
Comment (PierreLeveau) [inline train.py:282]: Would it be difficult to put  the code in a `PytorchVisionImageClassificationModel`  class to write:
```
training_loss = PytorchVisionImageClassificationModel(
                project_id, api_key, api_endpoint
            ).train(
                assets=assets,
                job=job,
                job_name=job_name,
                model_framework=model_framework,
                model_name=model_name,
                clear_dataset_cache=clear_dataset_cache,
                disable_wandb=disable_wandb,
            )
```
? If this is, we will keep it as is an create a high priority technical debt ticket.

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
missing e2e test

File: predict.py [hunk 11]
```diff
@@ -146,6 +149,24 @@ def predict_one_job(
             prioritization,
             from_project=from_project,
         )
+    elif (
+        content_input == ContentInput.Radio
+        and input_type == InputType.Image
+        and ml_task == MLTask.Classification
+    ):
+        job_predictions: JobPredictions = train_and_get_error_image_classification(  # type: ignore
```
[the remark this answers is not in the document]
Comment (crsegerie) [reply predict.py:157]: I didn't saw it because of the  # type: ignore, but yes

File: kiliautoml/models/_hugging_face_text_classification_model.py [hunk 12]
```diff
@@ -26,54 +27,71 @@
...
 
     ml_task: MLTaskT = MLTask.Classification  # type: ignore
 
-    def __init__(self, project_id: str, api_key: str, api_endpoint: str) -> None:
+    def __init__(
+        self,
+        project_id: str,
+        api_key: str,
+        api_endpoint: str,
+        model_repository: ModelRepositoryT = "huggingface",
```
Comment (PierreLeveau) [inline kiliautoml/models/_hugging_face_text_classification_model.py:42]: `model_repository` is not variable, so this should be a constant attribute.

File: kiliautoml/utils/cleanlab/train_cleanlab.py [hunk 13]
```diff
@@ -161,138 +62,144 @@ def combine_folds(data_dir, model_dir, verbose=0, num_classes=10, nb_folds=4, se
...
-        train_idx, val_idx = train_test_split(cv_train_idx, test_size=0.2)
-        image_datasets = copy.deepcopy(original_image_datasets)
-        holdout_dataset = copy.deepcopy(image_datasets["val"])
-        holdout_dataset.imgs = [holdout_dataset.imgs[i] for i in cv_holdout_idx]
-        holdout_dataset.samples = holdout_dataset.imgs
-        image_datasets["train"].imgs = [image_datasets["train"].imgs[i] for i in train_idx]
-        image_datasets["train"].samples = image_datasets["train"].imgs
-        image_datasets["val"].imgs = [image_datasets["val"].imgs[i] for i in val_idx]
-        image_datasets["val"].samples = image_datasets["val"].imgs
+class ImageClassificationModel(BaseModel):
```
Comment (PierreLeveau) [inline kiliautoml/utils/cleanlab/train_cleanlab.py:70]: to move to its own file in `kiliautoml.models`

Comment (PierreLeveau) [reply kiliautoml/utils/cleanlab/train_cleanlab.py:70]: rename it to `PyTorchVisionImageClassificationModel`

Comment (PierreLeveau) [reply kiliautoml/utils/cleanlab/train_cleanlab.py:70]: and I know it is not ideal, but put the arguments in the same order as the HuggingFace ones

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
Good improvement. A few changes left. Can you put all the `.pkl` files into a `.zip` file (that would be extracted at testing time), and remove the individual files ?

File: kiliautoml/models/_hugging_face_named_entity_recognition_model.py [hunk 14]
```diff
@@ -36,42 +37,60 @@ class KiliNerAnnotations(TypedDict):
...
+        self,
+        project_id: str,
+        api_key: str,
+        api_endpoint: str,
+        model_name: ModelNameT = "bert-base-multilingual-cased",
+    ) -> None:
         KiliTextProjectMixin.__init__(self, project_id, api_key, api_endpoint)
         BaseModel.__init__(self)
 
+        model_name_setted: ModelNameT = set_default(  # type: ignore
```
Comment (PierreLeveau) [inline kiliautoml/models/_hugging_face_named_entity_recognition_model.py:52]: model_name_set
